fix: test_web_browse_permission_denied が NoneType TypeError で失敗 (Python 3.12 CI)

## 概要

CI の `test (3.12)` が1件失敗している。

## 失敗テスト

`tests/test_agentcore.py::test_web_browse_permission_denied`

## エラー内容

```
E   TypeError: argument of type 'NoneType' is not iterable
```

テスト内の `assert "No permission" in result` で `result` が `None` を返している。

## 発生箇所

- File: `tests/test_agentcore.py:58`
- Run: https://github.com/m4minidesk-sys/yui-agent/actions/runs/22559026513/job/65341721105

## 影響

PR #100 マージ済み後もCIが FAILURE のまま (864 passed / 1 failed)

## 修正方針

1. `web_browse` のパーミッション拒否時の戻り値を確認
2. `None` を返している場合はエラーメッセージ文字列を返すよう修正
3. またはテスト側で `result is not None` を事前アサートする

## 優先度

中（CI グリーンの維持）

Closes #101 とは無関係、別Issue。

--- a/src/yui/tools/agentcore.py
+++ b/src/yui/tools/agentcore.py
@@ -105,6 +105,9 @@ def web_browse(url: str, task: str = "extract main content", timeout: int = 30) 
                     logger.warning("Browser session cleanup failed (result already obtained): %s", ctx_e)
                 else:
                     raise
+            else:
+                # No result obtained yet (e.g. __enter__ failed) — propagate to outer handler
+                raise
         if _browse_result is not None:
             return _browse_result
 

